src/audio/volume/volume_generic_with_peakvol.c

```diff
@@ -338,15 +339,16 @@ static void vol_passthrough_s16_to_s16(struct processing_module *mod,
 	struct vol_data *cd = module_get_private_data(mod);
 	struct audio_stream *source = bsource->data;
 	struct audio_stream *sink = bsink->data;
-	int32_t *x, *x0;
-	int32_t *y, *y0;
+	int16_t *x, *x0;
+	int16_t *y, *y0;
```

Comment:
this changes the whole pointer arithmetic in loops below, are we sure it wasn't 32-bit on purpose? @singalsu

Reply:
Precisely if I change the variable type to int16_t and add a cast to the function then the test works correctly. 

The test I am fixing is this
test_09_peak_vol.py::TestPeakVolFull::test_01_09_peakvol_quality[44100Hz_16in16bit_1ch-0]

Reply:
> this changes the whole pointer arithmetic in loops below, are we sure it wasn't 32-bit on purpose? @singalsu

I think that's a bug and this be untested code since we haven't seen it.

Reply:
I was also thinking first this would be an optimization to handle two s16 samples with same load/store but it's likely some unfinished broken optimization or just a plain mistake with pointer type.

Reply:
> As far as I can see, changing the variable type to int16_t and not adding a cast also works correctly

@gbernatxintel yes, typecasts aren't needed, please drop